release_dashboard/lib/widgets/common/dialog_prompt.dart

```diff
@@ -4,37 +4,113 @@
 
 import 'package:flutter/material.dart';
 
-/// Widget that prompts an alert dialogue to the current window and forces the user to choose between two options.
+/// Function that prompts an alert to the current window and forces the user to choose between two options.
 Future<String?> dialogPrompt({
   required BuildContext context,
-  required String title,
-  required String content,
-  required String leftOptionTitle,
-  required String rightOptionTitle,
-  VoidCallback? leftOptionCallback,
-  VoidCallback? rightOptionCallback,
+  required Widget title,
+  required Widget? content,
+  required String leftButtonTitle,
+  required String rightButtonTitle,
+  VoidCallback? leftButtonCallback,
+  VoidCallback? rightButtonCallback,
 }) {
   return showDialog<String>(
     context: context,
     builder: (BuildContext context) => AlertDialog(
-      title: Text(title),
-      content: Text(content),
+      title: title,
+      content: content,
       actions: <Widget>[
         TextButton(
           onPressed: () {
-            if (leftOptionCallback != null) leftOptionCallback();
-            Navigator.pop(context, leftOptionTitle);
+            if (leftButtonCallback != null) leftButtonCallback();
+            Navigator.pop(context, leftButtonTitle);
           },
-          child: Text(leftOptionTitle),
+          child: Text(leftButtonTitle),
         ),
         TextButton(
           onPressed: () {
-            if (rightOptionCallback != null) rightOptionCallback();
-            Navigator.pop(context, rightOptionTitle);
+            if (rightButtonCallback != null) rightButtonCallback();
+            Navigator.pop(context, rightButtonTitle);
           },
-          child: Text(rightOptionTitle),
+          child: Text(rightButtonTitle),
         ),
       ],
     ),
   );
 }
+
+/// Widget that prompts an alert to the current window and forces the user to choose between two options.
+///
+/// User also has to input [confirmationString] as an extra layer of validation.
+/// The right button becomes enabled when [_userInput] matches [confirmationString], otherwise it is disabled.
+class DialogPromptInputConfirm extends StatefulWidget {
+  const DialogPromptInputConfirm({
+    required this.title,
+    required this.content,
+    required this.leftButtonTitle,
+    required this.rightButtonTitle,
+    required this.confirmationString,
+    this.leftButtonCallback,
+    this.rightButtonCallback,
+    Key? key,
+  }) : super(key: key);
+
+  final Widget title;
+  final Widget? content;
+  final String leftButtonTitle;
+  final String rightButtonTitle;
+  final String confirmationString;
```

Comment:
dartdoc?

Reply:
I have one [here](https://github.com/Yugue/cocoon/blob/3cc882dbe85412c83f7d8a43254001b808ce55e7/release_dashboard/lib/widgets/common/dialog_prompt.dart#L42). Is this enough?

Reply:
All fields should have dart docs

Reply:
An example would be

```
  /// [String] user is expected to enter to enable [rightButtonCallback].
  final String confirmationString;
```